include/bluetooth/bluetooth.h

```diff
@@ -389,6 +389,9 @@ enum {
 	 *  advertising data.
 	 *  If the GAP device name does not fit into advertising data it will be
 	 *  converted to a shortened name if possible.
+	 *  @ref BT_LE_ADV_OPT_FORCE_NAME_IN_AD can be used to force the device
```

Comment:
We have have had this discussion a few times already, and the last agreement was to document how the application can do this themselves instead.
It is documented below.

Last issue here: https://github.com/zephyrproject-rtos/zephyr/issues/27692#issuecomment-740693534

@jhedberg @Thalley  Do we wish to revert this decision then, it keeps getting requested so makes sense to listen to it.

But then I think we should remove the below statement again.

Reply:
It's a very good question. If it indeed does get keep requested, that's a good indication that the API can be improved (e.g. by the result of this PR). 

I am a bit reluctant to add this, as it is already fairly easy for the application. We even have:

```
	 *  The application can set the device name itself by including the
	 *  following in the advertising data.
	 *  @code
	 *  BT_DATA(BT_DATA_NAME_COMPLETE, name, strlen(name))
	 *  @endcode
```

In the documentation for `BT_LE_ADV_OPT_USE_NAME` (which I also originally missed), which is even being used in the `beacon` and `peripheral_ots` samples. Maybe there's a way we can make that even clearer somehow? 

@lairdjm Given this, are you still in favor of adding this option from your PR?

Reply:
I am in favor of having my PR accepted, I prefer the option so that it can be supplied like this.